valhalla/midgard/shape_decoder.h

```diff
@@ -52,8 +51,7 @@ template <typename Point> class Shape5Decoder {
   Point pop() noexcept(false) {
     lat = next(lat);
     lon = next(lon);
-    return Point(typename Point::first_type(double(lon) * prec),
-                 typename Point::second_type(double(lat) * prec));
+    return Point(double(lon) * prec, double(lat) * prec);
```

Comment:
by not doing explicit conversion here we can allow for more pair/tuple-like objects to be used with this function. it allows you to bring your own point class basically.